framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -417,19 +418,61 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
             }
         }
 
-        // Check type-system specific annotations.
-        Class<PostconditionAnnotation> metaAnnotation = PostconditionAnnotation.class;
+        // Check for type-system specific annotations.
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
-                factory.getDeclAnnotationWithMetaAnnotation(methodElement, metaAnnotation);
+                factory.getDeclAnnotationWithMetaAnnotation(
+                        methodElement, PostconditionAnnotation.class);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
+
+            result.addAll(getPostConditionAnnotations(anno, metaAnno));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a collection of postcondition annotations that can consist of
+     * {@code @EnsuresNonNull},{@code @EnsuresKeyFor},{@code @EnsuresLTLengthOf},
+     * {@code @EnsuresLockHeld} or any other postcondition annotation developed in the future.
+     *
+     * @param anno wrapper annotation if multiple annotations are present, otherwise single
```

Comment:
What kind of single annotation?  This is a place where type information would be useful.

Reply:
@mernst What do you mean by `type information` here?
Thanks

Reply:
Information about the type of annotation represented by `anno`.  Might that be (a wrapper for) *any* annotation?  Or should clients only pass certain types of arguments?

Reply:
I have updated about the type of annotation(i.e. postcondition annotation) and that clients should pass only postcondition annotation. Please review it.